components/src/dynamo/planner/connectors/kubernetes.py

```
@@ -311,6 +311,134 @@ def get_gpu_counts(
+        has no matching component — handled gracefully by the caller).
+
+        ``deployment`` lets a caller that already fetched the DGD this tick
+        (e.g. a power-annotation sweep resolving both prefill and decode) reuse
+        that snapshot, avoiding a redundant apiserver GET per role. When None,
+        the current DGD is fetched.
+        """
+        if deployment is None:
+            deployment = self.kube_api.get_graph_deployment(self.graph_deployment_name)
+        try:
+            service = get_component_from_type_or_name(deployment, sub_component_type)
+        except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)